Add our path resolution policy to developing (#45)
fox-it/dissect-docs: docs/source/contributing/developing.rst

- :doc:`Tooling </contributing/tooling>`: information regarding the available tooling for building and testing code and documentation.

Path resolution policy
----------------------

> Starting from Dissect version 3.11, we decided that new plugins will only provide the original parsed ``path`` value in its records.

Reviewer: ```suggestion
Starting from Dissect version 3.11, the policy is for plugins to only provide the original parsed ``path`` value in returned records.
```

> We recommend external contributors to follow the same policy.
> This keeps the information inside the records as close to the truth as possible.

Reviewer: ```suggestion
We recommend external contributors to follow the same policy,
this keeps the path information in the records as close to the truth as possible.
```

> A resolved path can still be accessed by using the ``--resolve`` or ``--hash`` flags in :doc:`target-query </tools/target-query>`.

Reviewer: ```suggestion
Resolved paths can be accessed by using the ``--resolve`` or ``--hash`` flags in :doc:`target-query </tools/target-query>`.
```

ese flags add additional fields to the record using the identifier/name of the field as a prefix making it easier to identify its origin.
As an example, lets say we have the following information inside a record:

.. code-block::

    some_name=%SystemRoot%\Temp\file

The ``--resolve`` adds, as the name, the resolved path to the record and transforms the information above to:

.. code-block::

    some_name=%SystemRoot%\Temp\file
    some_name_resolved=C:\Windows\Temp\file

> While the ``--hash`` flag adds extends the ``--resolve`` information and includes the ``digest`` to it.

Reviewer 2: ```suggestion
The ``--hash`` flag extends the ``--resolve`` information and includes a ``_digest`` suffixed field as well, which contains the file digest triple (if available):
```

Reviewer 2: The `if available` was a reference to if the file exists, but I've reworded it.
